packages/transport/src/modules/relationshipTemplates/RelationshipTemplateController.ts

```diff
@@ -246,4 +248,10 @@ export class RelationshipTemplateController extends TransportController {
 
         return relationshipTemplate;
     }
+
+    public async cleanupTemplateOfDecomposedRelationship(template: RelationshipTemplate): Promise<void> {
+        if (!template.isOwn || template.cache?.maxNumberOfAllocations === 1) {
+            await this.templates.delete(template);
+        }
```

Comment:
maybe we should remove *all* templates of the peer - I don't care if this happens in another method or in this one (which should then be renamed)

Reply:
why? We are decomposing *here* to get rid of all data.

Reply:
just from the perspective that you delete something once it's useless - it depends on whether you want to create a relationship with an old template or if the history is of any use

Reply:
It's not only about sth to be useless, it's about that that Templates are containing sensitive data.